src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -111,14 +117,75 @@ public override bool Execute()
             // Update framework dependencies sections
             for (int i = 0; i < Frameworks.Length; i++)
             {
-                dependencies = GenerateDependencies(projectRoot, ExternalPackages, Frameworks[i]);
+                dependencies = GenerateDependencies(projectRoot, ExternalPackages, packageInformation, Frameworks[i]);
                 projectRoot = UpdateDependenciesProperty(projectRoot, dependencies, Frameworks[i]);
             }
             WriteProject(projectRoot, OutputProjectJson);
 
             return true;
         }
 
+        /// <summary>
+        /// Given a package name regex pattern, and an array of drop locations, create an array of objects
+        /// containing package information (name, version,, prerelease version)
+        /// </summary>
+        /// <param name="packagesDrops"></param>
+        /// <returns></returns>
+        private IEnumerable<ITaskItem> GatherPackageInformationFromDrops(string [] packagesDrops)
+        {
+            List<ITaskItem> packageNameItems = new List<ITaskItem>();
+
+            foreach (string packageDrop in packagesDrops)
+            {
+                if (!Directory.Exists(packageDrop))
+                {
+                    Log.LogWarning("PackageDrop does not exist - '{0}'", packageDrop);
+                    continue;
+                }
+                IEnumerable<ITaskItem> packages = Directory.GetFiles(packageDrop).Select(f => new TaskItem(Path.GetFileNameWithoutExtension(f)));
+
+                foreach (ITaskItem package in packages)
+                {
+                    packageNameItems.Add(CreatePackageItemFromString(package.ItemSpec));
```

Comment:
@chcosta take a look at *PackageArchiveReader* and *NuspecReader* from *NuGet.Packaging*

NuGet does some parsing on filenames also, but it is imperfect and should only be used to rule out obvious non-matches. For example packageA.1.2.3 could be either: packageA.1 version: 2.3 or packageA version: 1.2.3. The only way to know for sure is to read the nuspec.

You can see the NuGet code for this here: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/LocalFolderUtility.cs#L264